Kernel/Heap/kmalloc.cpp

```diff
@@ -182,17 +183,67 @@ struct KmallocGlobalHeap {
     }
 };
 
+[[gnu::malloc, gnu::returns_nonnull, gnu::alloc_size(1)]] void* kmalloc_static_eternal(size_t);
+
+class EternalHeap {
+    AK_MAKE_NONCOPYABLE(EternalHeap);
+    AK_MAKE_NONMOVABLE(EternalHeap);
+
+public:
+    void* operator new(size_t size) { return kmalloc_static_eternal(size); }
```

Comment:
This doesn't feel good. We now have a dedicated function called `kmalloc_static_external` for bump-allocating the `EternalHeap` objects, and these objects magically end up in a the `.heap` ELF segment.

I think the code would be a lot easier to understand if we allocated the memory for them, and then used placement new to construct the `EternalHeap` (instead of overriding `operator new`)